x/auth/client/cli/cli_test.go

```diff
@@ -985,6 +986,50 @@ func (s *IntegrationTestSuite) TestTxWithoutPublicKey() {
 	s.Require().NoError(err)
 }
 
+func (s *IntegrationTestSuite) TestFlagOutputDocument() {
+	val0 := s.network.Validators[0]
+	res, err := bankcli.MsgSendExec(
+		val0.ClientCtx,
+		val0.Address,
+		val0.Address,
+		sdk.NewCoins(
+			sdk.NewCoin(fmt.Sprintf("%stoken", val0.Moniker), sdk.NewInt(10)),
+			sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10)),
+		),
+		fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10))).String()),
+		fmt.Sprintf("--%s=true", flags.FlagGenerateOnly),
+	)
+	s.Require().NoError(err)
+
+	// Write  unsigned tx to file.
+	unsignedTx, cleanup1 := testutil.WriteToNewTempFile(s.T(), res.String())
+	defer cleanup1()
+
+	// Create a temp file to hold the signed tx.
+	tempfile, err := ioutil.TempFile("", "testtx")
+	s.Require().NoError(err)
+	err = tempfile.Chmod(0644)
+	s.Require().NoError(err)
+	err = tempfile.Close()
+	s.Require().NoError(err)
+	defer func() {
+		err = os.Remove(tempfile.Name())
+		s.Require().NoError(err)
```

Comment:
This may cause the test to fail if the temporary file is not deleted successfully at the end of the test (which might happen for a number of reasons that are entirely unrelated to the test case itself)

Reply:
ah, what are best practices here? should I just ignore the err?

Reply:
I do not know if I agree with this. I would keep this as-is. If it errors, so be it. Just rerun the test. Ignoring errors in tests is typically not something you want to do.

Reply:
This is about the removal of a temporary file that is created within the execution of a test. If we want to go for The Right Way(c), then yes we might want to cause the test fail. Though weird things happen in a CI that we don't fully control, e.g. resources might become unavailable before the program exits. In my opinion, it'd just be a self-inflicted pain in the lower back if we had to go and manually re-run all tests because of a glitch in the CI. Thus we should just ignore the error and let go.

Reply:
Solved with https://github.com/cosmos/cosmos-sdk/pull/8121#discussion_r539490215